apps/fz_http/lib/fz_http_web/live/setting_live/oidc_form_component.ex

```diff
@@ -174,55 +175,50 @@ defmodule FzHttpWeb.SettingLive.OIDCFormComponent do
       assigns.providers
       |> Map.get(assigns.provider_id, %{})
       |> Map.put(:id, assigns.provider_id)
-      |> FzHttp.Configurations.Configuration.OpenIDConnectProvider.changeset()
+      |> Configurations.Configuration.OpenIDConnectProvider.create_changeset()
 
-    {:ok,
-     socket
-     |> assign(assigns)
-     |> assign(:external_url, FzHttp.Config.fetch_env!(:fz_http, :external_url))
-     |> assign(:changeset, changeset)}
+    socket =
+      socket
+      |> assign(assigns)
+      |> assign(:external_url, FzHttp.Config.fetch_env!(:fz_http, :external_url))
+      |> assign(:changeset, changeset)
+
+    {:ok, socket}
   end
 
   def handle_event("save", %{"open_id_connect_provider" => params}, socket) do
-    changeset =
-      params
-      |> FzHttp.Configurations.Configuration.OpenIDConnectProvider.changeset()
-      |> render_changeset_errors()
-
-    update =
-      case changeset do
-        %{valid?: true} ->
-          {:ok, _} =
-            changeset
-            |> Ecto.Changeset.apply_changes()
-            |> Map.from_struct()
-            |> Map.new(fn {k, v} -> {to_string(k), v} end)
-            |> then(fn data ->
-              id = Map.get(data, "id")
-
-              %{
-                openid_connect_providers:
-                  socket.assigns.providers
-                  |> Map.delete(socket.assigns.provider_id)
-                  |> Map.put(id, data)
-                  |> Map.values()
-              }
-            end)
-            |> FzHttp.Configurations.update_configuration()
-
-        _ ->
-          {:error, changeset}
-      end
-
-    case update do
-      {:ok, _config} ->
-        {:noreply,
-         socket
-         |> put_flash(:info, "Updated successfully.")
-         |> redirect(to: socket.assigns.return_to)}
-
-      {:error, changeset} ->
-        {:noreply, assign(socket, :changeset, changeset)}
+    create_changeset = Configurations.Configuration.OpenIDConnectProvider.create_changeset(params)
```

Comment:
Will this create duplicate config entries?

Reply:
Nope, its just a hack to validate params which was already there before PR

Reply:
KK 👍